src/main/java/org/jabref/gui/FXDialogService.java

```diff
@@ -253,7 +270,8 @@ public boolean showConfirmationDialogWithOptOutAndWait(String title, String cont
 
     @Override
     public void notify(String message) {
-        JabRefGUI.getMainFrame().output(message);
+        LOGGER.info(message);
+        DefaultTaskExecutor.runInJavaFXThread(() -> statusLine.fireEvent(new SnackbarEvent(new JFXSnackbarLayout(message), TOAST_MESSAGE_DISPLAY_TIME, null)));
```

Comment:
Please remove the `runInJavaFXThread`. This is a leftover when we had a lot of interaction between Swing and JavaFX code.

Reply:
I think this method is still sometimes called from background threads therefore we risk exceir

Reply:
Is there an alternative to this method? I also think it is necessary to run it on the JavaFX thread, for example in `SendAsEmailAction.action` `notify` is called from a background thread.

Reply:
Usually background tasks are not allowed to communicate things directly. The general principle is that only the hard work is moved to a different task (by using `BackgroundTask.wrap(doHardWork)`) and the only user interaction should be in the `onSuccess` and `onError` handlers that automatically run on the JavaFX thread).

Personally, I would prefer exceptions instead of facing potentially hard to debug threading issues. But for now you can simply leave the `runInJavaFXThread` call.

Reply:
Ok, I understand and will keep that in mind.